lib/cn/cn_tree.c

```diff
@@ -1576,13 +1597,16 @@ cn_comp_commit(struct cn_compaction_work *w)
     }
 
     kvsets = calloc(1, alloc_len);
-    if (ev(!kvsets)) {
-        w->cw_err = merr(ENOMEM);
+    cookiev = calloc(num_out, sizeof(*cookiev));
+
+    if (!kvsets || !cookiev) {
+        err = merr(ENOMEM);
```

Comment:
At some point we need to move stuff like these allocations which can fail from commit to prepare..